src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,503 @@ title: Builder.io & Astro
+
+2. Go to the **Models** tab in your Builder space, pick the model you've created and paste the URL from step 1 into the **Preview URL** field. Make sure the URL is complete and includes the protocol, for example `https://`.
+
+3. Click the **Save** button in the upper right.
+
+:::tip
+When you deploy your site, change the preview URL to match your production URL, for example `https://myAwesomeAstroBlog.com/builder-preview`.
+:::
+
+#### Testing the preview URL setup
+
+1. Make sure your site is live (e.g. your dev server is running) and the `/builder-preview` route is working.
+
+2. In your Builder space under the **Content** tab, click on **New** to create a new content entry for your `blogpost` model.
+
+3. In the Builder editor that just opened, you should be able to see the `builder-preview.astro` page with a big **Add Block** in the middle.
+
+:::tip[Troubleshooting]
+
+Things can sometimes go wrong when setting up the preview. If something's not right, you can try one of these things:
+
+* Make sure the site is live - for example, your dev server is running.
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+* Make sure it's the full URL including the protocol, for example `https://`.
+* If you're working in a virtual environment like [StackBlitz](https://stackblitz.com/) or [Gitpod](https://www.gitpod.io/), you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read [Builder's troubleshooting guide](https://www.builder.io/c/docs/guides/preview-url-working).
+:::
+
+### Creating a blog post
+
+1. Before you can create new content in Builder, make sure [the preview URL for your model is working](#setting-up-the-preview).
+2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-setup), you can use that entry. Otherwise, create a new one: Go to the **Content** tab in your Builder space and click on **New**.
+3. In the panel on the right under the **options** tab, you should see the data fields `title` and `slug`. These are the fields you've created when you [set up your model](#creating-a-model-for-a-blog-post). Fill them in with the following values:
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+4. Design your post. Click on the **Add Block** button and add something - maybe a text field with some text to start with.
+5. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+6. When you're ready click the **Publish** button in the upper right corner.
+
+Your content is now live in the Builder CMS but not part of your Astro project yet. You'll be fetching and displaying it in the following steps.
```

Comment:
```suggestion
1. In Builder's visual editor, create a new content entry with the following values:
    - **title:** 'First post, woohoo!'
    - **slug:** 'first-post-woohoo'
2. Complete your post using the **Add Block** button and add a text field with some post content .
3. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
4. When you're ready click the **Publish** button in the upper right corner.
5. Create as many posts as you like, ensuring that all content entries contain a `title` and a `slug` as well as some post content.
```
Again, trying not to overload with too much Builder-specific instructions but rather to emphasize creating entries that will match the desired schema.

Also, I removed the link to the section immediately proceeding, since we can assume they've just done that and just been there!

In fact, linking to various steps throughout is generally discouraged because you expect, and want, people to follow your instructions linearly and not be jumping around too much. There is a sidebar on the right if they really need it. But otherwise, the goal is just to get them chugging through, top to bottom. We don't need to check that they did an earlier step. That's on them! lol